pkgs/os-specific/windows/msvcSdk/default.nix

```diff
@@ -0,0 +1,132 @@
+{
+  lib,
+  config,
+  stdenvNoCC,
+  xwin,
+  testers,
+  llvmPackages,
+}:
+let
+  version = (builtins.fromJSON (builtins.readFile ./manifest.json)).info.productSemanticVersion;
+
+  hashes = (builtins.fromJSON (builtins.readFile ./hashes.json));
+
+  host = stdenvNoCC.hostPlatform;
+  arch =
+    if host.isx86_64 then
+      "x86_64"
+    else if host.isAarch64 then
+      "aarch64"
+    else if host.isx86_32 then
+      "x86"
+    else if host.isAarch32 then
+      "aarch"
+    else
+      throw "Unsupported system";
+in
+if !config.microsoftVisualStudioLicenseAccepted then
+  throw ''
+    Microsoft Software License Terms are not accepted with config.microsoftVisualStudioLicenseAccepted.
+    Please read https://visualstudio.microsoft.com/license-terms/mt644918/ and if you agree, change your
+    config to indicate so.
+  ''
+else
+  stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit version;
+    pname = "msvc-sdk";
+    dontUnpack = true;
+
+    strictDeps = true;
+    nativeBuildInputs = [ xwin ];
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = hashes.${arch};
+
+    __structuredAttrs = true;
+    xwinArgs = [
+      "--accept-license"
+      "--cache-dir=xwin-out"
+      "--manifest=${./manifest.json}"
+      "--arch=${arch}"
+      "splat"
+      "--preserve-ms-arch-notation"
+      "--use-winsysroot-style"
```

Comment:
Ah this is what combines them? Yes, I rather not do that.

Reply:
`"--use-winsysroot-style"`? Yeah it combines them to how Microsoft does it on windows. Without that flag it just requires how it was in a previous commit, you can see the diff here on lines 104 and 105
https://github.com/NixOS/nixpkgs/compare/2086f8daee6bfc3c693548b12f577ca09cb18283..bee4f2aa104f2af639b88333c0d89cc63fb83e36#diff-a6b20d16ecce6e45aa55f20a50a53706a55bfdf20ec12ffd1842c929e3e60003L103-L106